keyboards/40percentclub/25/25.h

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2018
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+#define ___ KC_NO
+
+#ifndef FLIP_HALF
+#define LAYOUT_split( \
+	L00, L01, L02, L03, L04, R00, R01, R02, R03, R04, \
+	L10, L11, L12, L13, L14, R10, R11, R12, R13, R14, \
+	L20, L21, L22, L23, L24, R20, R21, R22, R23, R24, \
+	L30, L31, L32, L33, L34, R30, R31, R32, R33, R34, \
+	L40, L41, L42, L43, L44, R40, R41, R42, R43, R44  \
+) { \
+	{ L00, L01, L02, L03, L04 }, \
+	{ L10, L11, L12, L13, L14 }, \
+	{ L20, L21, L22, L23, L24 }, \
+	{ L30, L31, L32, L33, L34 }, \
+	{ L40, L41, L42, L43, L44 }, \
+	{ R04, R03, R02, R01, R00 }, \
+	{ R14, R13, R12, R11, R10 }, \
+	{ R24, R23, R22, R21, R20 }, \
+	{ R34, R33, R32, R31, R30 }, \
+	{ R44, R43, R42, R41, R40 }  \
+}
+#else
+#define LAYOUT_split( \
+	L00, L01, L02, L03, L04, R00, R01, R02, R03, R04, \
+	L10, L11, L12, L13, L14, R10, R11, R12, R13, R14, \
+	L20, L21, L22, L23, L24, R20, R21, R22, R23, R24, \
+	L30, L31, L32, L33, L34, R30, R31, R32, R33, R34, \
+	L40, L41, L42, L43, L44, R40, R41, R42, R43, R44  \
+) { \
+	{ L00, L01, L02, L03, L04 }, \
+	{ L10, L11, L12, L13, L14 }, \
+	{ L20, L21, L22, L23, L24 }, \
+	{ L30, L31, L32, L33, L34 }, \
+	{ L40, L41, L42, L43, L44 }, \
+	{ R00, R01, R02, R03, R04 }, \
+	{ R10, R11, R12, R13, R14 }, \
+	{ R20, R21, R22, R23, R24 }, \
+	{ R30, R31, R32, R33, R34 }, \
+	{ R40, R41, R42, R43, R44 }  \
+}
+#endif
+
+#define LAYOUT_macro( \
+	L00, L01, L02, L03, L04, \
+	L10, L11, L12, L13, L14, \
+	L20, L21, L22, L23, L24, \
+	L30, L31, L32, L33, L34, \
+	L40, L41, L42, L43, L44  \
+) { \
+	{ L00, L01, L02, L03, L04 }, \
+	{ L10, L11, L12, L13, L14 }, \
+	{ L20, L21, L22, L23, L24 }, \
+	{ L30, L31, L32, L33, L34 }, \
+	{ L40, L41, L42, L43, L44 }, \
+	{ ___, ___, ___, ___, ___ }, \
+  { ___, ___, ___, ___, ___ }, \
+  { ___, ___, ___, ___, ___ }, \
+  { ___, ___, ___, ___, ___ }, \
+  { ___, ___, ___, ___, ___ }  \
+}
+
+#define LAYOUT LAYOUT_macro
```

Comment:
This isn't needed here.

```suggestion
```

Reply:
Given a quick grep across the code base, `grep -h -r "#define LAYOUT" keyboards/ | sort | uniq -c | sort -k 1 -n` gives 192 instances of `#define layout` and at least 20 instances of aliases. It was an attempt to make the keymaps more consistent with other boards and potentially provide a clearer intention within the keymap files.

With the other suggested changes, my preference would be to go with something like:
`#define LAYOUT_macro LAYOUT_ortho_5x5
#define LAYOUT_split LAYOUT_ortho_5x10
#define LAYOUT LAYOUT_ortho_5x5`

Reply:
That may be, but if you note, a lot of those boards don't match a specific layout.  Which is why they use `LAYOUT` and not a distinct name. 

Additionally, when there is a matching macro name, we want to use that instead of just `LAYOUT`, so that it is easier to convert for newer users.   
You can see a list of existing layouts (not comprehensive) here: https://github.com/qmk/qmk_firmware/tree/master/layouts/default

Also, the macro names do show up in the QMK Configurator, so they should be meaningful and descriptive.  Which is a factor, as well. 

And yes, that should be fine:
```suggestion
#define LAYOUT_macro LAYOUT_ortho_5x5 #define LAYOUT_split LAYOUT_ortho_5x10 #define LAYOUT LAYOUT_ortho_5x5
```

Reply:
@drashna 

> Also, the macro names do show up in the QMK Configurator, so they should be meaningful and descriptive. Which is a factor, as well.

To a point, but if you assign an alias for a macro, the alias name shows up in the Configurator, rather than the direct name.

This:

```
#define LAYOUT_ortho_5x5( \
  L00, L01, L02, L03, L04, \
  L10, L11, L12, L13, L14, \
  L20, L21, L22, L23, L24, \
  L30, L31, L32, L33, L34, \
  L40, L41, L42, L43, L44  \
) { \
  { L00, L01, L02, L03, L04 }, \
  { L10, L11, L12, L13, L14 }, \
  { L20, L21, L22, L23, L24 }, \
  { L30, L31, L32, L33, L34 }, \
  { L40, L41, L42, L43, L44 }  \
}

#define LAYOUT LAYOUT_ortho_5x5
```

will show up in the Configurator as `LAYOUT` rather than `LAYOUT_ortho_5x5`, which itself won't be shown as an option.